src/pages/Index.vue

```diff
@@ -136,12 +155,21 @@ export default defineComponent({
       }
     },
 
-    serveAtack (_event: unknown, data: { url: string, log: string }) {
+    serveAtack (_event: unknown, data: { url: string, log: string, success: boolean }) {
       if ((new Date()).getTime() - this.lastAtackChange.getTime() > 1000) {
         this.currentAtack = data.url
         this.lastAtackChange = new Date()
       }
+
+      // Counters
       this.atackCounter += 1
+      if (!data.success) this.atackErrorCounter += 1
+
+      // Latest 500 requests success/failure - used for IP change suggestion
+      if (this.latest500Responses.length > 500) this.latest500Responses.shift()
+      this.latest500Responses.push(data.success)
```

Comment:
Хм, мне кажется для большого к-ва "тредов" может работать достаточно так себе(в случае если за раз тебе будет прилетать 500+ ответов) - да и в целом вопрос на счет UI не будет ли он глючить, например, при 1к "раннеров". Всё же с каждым изменением приходится прогонять `shift, filter`, а это +O(n) на ровном месте и иногда это вообще не нужно(для runnersNumber > 500)
Я бы тут лучше немного сменил эту логику, чтобы не использовать массивы, а вручную считать, что-то ака
```
// state
lastMinuteStatus: { success: 0, failed: 0 },
currentStatus: { success: 0, failed: 0 }
// methods
serveAtack(..) {
...
  if (data.success){
    currentStatus.successful =+ 1
    lastMinuteStatus.successful += 1
  } else {
    currentStatus.failed += 1
    lastMinuteStatus.failed += 1
  }
  ...
  },
  
  // todo: naming
checkStatus() {
 if (lastMinuteStatus.successful < lastMinuteStatus.failed) {
    // suggest IP change
    lastMinuteStatus.successful = 0
    lastMinuteStatus.failed = 0
 },
}
...
mounted() {
   setTimeout(() => { this.checkStatus() }, 60 * 1000)
}

Reply:
*если что, то отображать статистику по `currentStatus `

Reply:
First of all for bigger amount of runners: please take a look how runners are implemented. Each runner has it's own random target - so the amount of runner doesn't impact the amount of simultaneous ECONNABORTED errors.

In regards to O(n) for an array of 500 elements - it is very very very fast.

In regards to rerender (UI глючить) - we are already rerendering after each response by incrementing counter and adding log line!

Your solution - once in a minute (not even in a minute because `setTimeout(() =>, 60000)` is not a minute) we will reset everything, so if first 10 requests will be failed we will need to show the message for IP change, but all other 90 will be ok - we will need to show the message that IP is fine.
We will end up having flashing text that will be changing once in a minute and will be confusing everyone for sure!

Reply:
PS: please use English in your reviews. The main contributor of this project is from Poland and obviously he doesn't understand Ukrainian 😉

Reply:
As far as i can see `latest500Responses` is aggregated for all requests(`attack` event is bubbled to the `doser` and after that we're sending it to UI).
Therefore technically for large `n` it can be more flashy than checking it each minute(or some other arbitrary number).
But anyway if you say it works good for "large" N(i'd say 256-1k), than you can resolve this thread :)